.github/workflows/node.js.yml

```diff
@@ -10,28 +10,45 @@ on:
         branches: [master, feature/*]
 
 jobs:
-    build:
-        name: Build
+    macos:
+        name: macOS nodejs
         runs-on: macos-latest
-
         strategy:
             matrix:
-                node-version: [10.x]
-
+                node-version: [12.x]
         steps:
             - uses: actions/checkout@v2
             - name: Use Node.js ${{ matrix.node-version }}
-              uses: actions/setup-node@v1
+              uses: actions/setup-node@v2
               with:
                   node-version: ${{ matrix.node-version }}
             - run: npm install
             - run: npm run vscode:prepublish
               env:
                   NODE_OPTIONS: '--max-old-space-size=8192'
-            - name: Run headless test
-              uses: GabrielBB/xvfb-action@v1.2
+            - name: Tests
+              uses: GabrielBB/xvfb-action@v1
               with:
                   run: npm test
             - name: Code coverage
               if: ${{ github.repository == 'aws/aws-toolkit-vscode' && ( github.ref == 'master' || github.event_name == 'pull_request' ) }}
               run: bash <(curl -s https://codecov.io/bash) -s .coverage
+
+    windows:
```

Comment:
The two jobs appear to be similar, except for codecov and xvfb. Is it possible to use a matrix and conditionally run those steps? The duplication seems more challenging to maintain

Reply:
I considered that but estimated that the matrix approach will be more complicated. E.g. the `uses: GabrielBB/xvfb-action@v1` is not applicable to Windows, and also the code coverage code is different for Windows. So several things would need to be parameterized and it might be less understandable.